class/Services/Content/Passle/PassleTagsContentService.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace Passle\PassleSync\Services\Content\Passle;
+
+use Exception;
+use Passle\PassleSync\Services\OptionsService;
+use Passle\PassleSync\Utils\UrlFactory;
+use Passle\PassleSync\Utils\Utils;
+
+class PassleTagsContentService
+{
+  public static function fetch_all()
+  {
+    $passle_shortcodes = OptionsService::get()->passle_shortcodes;
+
+    /** @var array[] $results */
+    $results = array_map(fn ($passle_shortcode) => static::fetch_all_by_passle($passle_shortcode), $passle_shortcodes);
+
+    if (in_array(true, array_map(function ($r) {
+      return is_wp_error($r);
+    }, $results))) {
+      dd($results);
+      throw new Exception("Failed to get data from the API", 500);
+    }
+
+    $result = array_merge(...$results);
+
+    if (!is_null($result)) {
+      return $result;
+    } else {
+      return array();
+    }
+  }
+
+  public static function fetch_all_by_passle(string $passle_shortcode)
+  {
+    $url = (new UrlFactory())
+      ->path("/tags/".$passle_shortcode)
```

Comment:
Should use simpler and more readable syntax for string interpolation
```suggestion
      ->path("/tags/$passle_shortcode")
```